### ISSUE TITLE
SARIF report is invalid

### ISSUE THREAD

**Issue by naokikimura** [securego/gosec#582]
### Summary 

The report output with the `-fmt sarif` option is not valid as a SARIF.

Due to this issue, the SARIF file output by gosec could not be uploaded to GitHub code scanning.

### Steps to reproduce the behavior

1. Checkout the source code
    ```bash
    git clone https://github.com/naokikimura/hello-golang.git
    cd hello-golang
    git checkout e28cd9f598a2ce06cc07d6bda0cd4eb8650fc1cf
    ```
2. Scan with `gosec` to create a SARIF report (Result: [gosec.sarif](https://github.com/securego/gosec/files/6054314/gosec.sarif.txt))
    ```bash
    curl -sSfL https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v2.6.1
    gosec -fmt sarif -out gosec.sarif ./...
    ```
3. Validate the SARIF report with `yajsv`
    ```bash
    go get github.com/neilpa/yajsv@v1.4.0
    curl -sSfLO https://raw.githubusercontent.com/schemastore/schemastore/master/src/schemas/json/sarif-2.1.0-rtm.4.json
    yajsv -s sarif-2.1.0-rtm.4.json gosec.sarif
    ```

Validation fails with the following message:
```
gosec.sarif: fail: runs.0.tool.driver.rules: array items[2,3] must be unique
1 of 1 failed validation
```

I get a similar error on GitHub. (Detailed log: [github_actions_log.txt](https://github.com/securego/gosec/files/6054281/github_actions_log.txt))

```
Error: Unable to upload "gosec.sarif" as it is not valid SARIF:
- instance.runs[0].tool.driver.rules contains duplicate item
```

### gosec version

- Version: 2.6.1
- Git tag: v2.6.1
- Build date: 2021-01-22T10:39:07Z

### Go version (output of 'go version')

go version go1.16 darwin/amd64

### Operating system / Environment

- ProductName:	macOS
- ProductVersion:	11.2.1
- BuildVersion:	20D74

### Expected behavior

Output a valid SARIF report

I expect it to output a SARIF report that GitHub code scanning accepts. 

Reference: [SARIF support for code scanning - GitHub Docs](https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/sarif-support-for-code-scanning)

### Actual behavior

Duplicate elements in `tool.driver.rules[]`

There are likely to be some other problems. It looks like there are duplicates in `results[]`.

**Comment by ccojocar** [comment]
Hi @naokikimura, Thanks for reporting this issue. There were some recent fixes in the sarif formatter. Can you also reproduce the issue with the latest version from master?

**Comment by naokikimura** [comment]
Hi @ccojocar, I tested it on the master branch (01ee764ed89c4d3520d583ce2c13637b65daef09). 

The issue has been resolved.

I hope this fix will be released soon. When will this be released?

**Comment by naokikimura** [comment]
#565 :eyes:

**Comment by felipe-avelar** [comment]
I was facing a similiar issue and was solved on PR #565, but it still needs docker image to be updated, that will be fine as soon as gosec suports go 1.16, as you can see at #578.
Maybe watching this issue could help you. 🙂

**Comment by ccojocar** [comment]
Fixed by https://github.com/securego/gosec/releases/tag/v2.7.0
